include/k4arecord/playback.hpp

```diff
@@ -14,11 +14,218 @@
 #include <chrono>
 #include <string>
 #include <vector>
+#include <memory>
 
 namespace k4a
 {
 
-/** \class playback playback.hpp
+/** \class data_block playback.hpp <k4arecord/playback.hpp>
+ * Wrapper for \ref k4a_playback_data_block_t
+ *
+ * \sa k4a_playback_data_block_t
+ */
+class data_block
+{
+private:
+    class data_block_ref
+    {
```

Comment:
I might be missing something, but it looks like we have two classes wrapping this to add refcounting semantics to a type that doesn't have them natively - why are we doing that?  We didn't for k4a_device_t; there, we just removed the copy constructors and only left a move constructor.

Reply:
Good question, I thought if I deleted the assignment operator I would loose the ability to pass the value to function. I like this option much better. Thank you!